src/Orthogonal/WeightFunction.jl

```diff
@@ -0,0 +1,206 @@
+SpecialPolynomials
+
+julia> w(t) = -log(t)
+w (generic function with 1 method)
+
+julia> SP.@register0 WF1 SP.WeightFunction   #  register  type WF1 as a weight function
+
+julia> SP.@register_weight_function WF1  MonicShiftedLegendre w
+
+julia> ν(j) = iszero(j) ? 1 : (-1)^j * gamma(j+1)^2 / (j*(j+1)*gamma(2j+1)) # help  out
+ν (generic function with 1 method)
+
+julia> SP.modified_moment(::Type{WF1},  j::Int) = ν(j)
+
+julia> αs, βs = -SP.Bn.(WF1, 0:5), SP.Cn.(WF1, 0:5);
+
+julia> [αs βs]
+6×2 Array{Float64,2}:
+ -0.75      1.0
+ -0.381148  0.423611
+ -0.504058  0.172646
+ -0.516236  0.203166
+ -0.517168  0.222419
+ -0.513854  0.239667
+```
+
+"""
+WeightFunction
+
+## We need w, π
+## w a function; π a *monic* system (ismonic(P) == true)
+weight_function(::Type{<:AbstractWeightFunction}) = throw(ArgumentError("No default method"))
+ϟ(P::Type{<:AbstractWeightFunction}) = throw(ArgumentError("No default method"))
+
+Polynomials.domain(W::Type{<:AbstractWeightFunction}) = (domain∘ϟ)(W)
+basis_symbol(::Type{<:AbstractWeightFunction}) = "W"
+
+
+
+## An,Bn,Cn for the monic orthogonal polynomials for the given weight function
+## We  have An=1;  α(n) =  -Bn(n); β(n) = Cn(n)
+An(::Type{W}, n) where {W <: AbstractWeightFunction} = one(eltype(W))
+
+function Bn(::Type{W}, k::Int) where {W <:AbstractWeightFunction}
+
+    P = ϟ(W)
+    
+    if k == 0
+        ν₀, ν₁ =  modified_moment(W, 0), modified_moment(W, 1)
+        out = Bn(P, 0) + ν₁ / ν₀
+    else
+        out = Bn(P, k) - sigma(W, k-1, k)/sigma(W, k-1, k-1)  + sigma(W, k, k+1)/sigma(W, k, k) 
+    end
+    -out
+    
+end
+
+function Cn(::Type{W}, k::Int) where  {W <: AbstractWeightFunction}
+    iszero(k) && return innerproduct(W, one, one)
+    (sigma(W, k, k)/sigma(W, k-1, k-1))
+end
+
+
+@memoize function sigma(W::Type{<:AbstractWeightFunction}, k, l) 
+
+    P = ϟ(W)
+    
+    T = eltype(W)
+    k == -1 && return zero(T)/1
+    k == 0  && return modified_moment(W, l) # ν_l = ∫π_l dw
+    k > l   && return zero(T)/1
+
+    # otherwise get through recursion (this uses Gautschi's formula, P&T have apparent error
+    sigma(W, k-1, l+1) - (Bn(W, k-1) - Bn(P, l)) * sigma(W, k-1, l) - (Cn(W, k-1))*sigma(W, k-2, l) + Cn(P,l)*sigma(W, k-1, l-1)
+    
+end
+
+## Compute the modified moment
+## v_j =  ∫ π_j ω(x) dx
+## (The moment is ∫ x^j ω(x) dx)
+## This can  be overridden when an exact calculuation is known, e.g.:
+function modified_moment(::Type{W},  j::Int) where {W <: AbstractWeightFunction}
+    P = ϟ(W)
+    @assert ismonic(P)
+    modified_moment(basis(P,j), W)
+end
+
+@memoize function modified_moment(πⱼ::AbstractOrthogonalPolynomial, W)
+    P = ϟ(W)
+    ω = weight_function(W)
+    πⱼdω = x -> πⱼ(x) * ω(x)
+    return ∫(πⱼdω, P, atol=sqrt(eps(float(eltype(W)))))
+end
+
+# integrate fdw over region specified by domain(P)
+function ∫(fdw, P; kwargs...)
+    dom = domain(P)
+    a, b = first(dom), last(dom)
+    if !first(Polynomials.inclusivity(dom))
+        a += eps(float(one(a)))
```

Comment:
What is this `± ε` code for?

(If you are worried about endpoint singularities, `quadgk` is documented to never evaluate the integrand exactly at the endpoints of the interval, which is useful precisely for that situation.)

Reply:
Thanks for the review! Very much appreciated. I'm not sure what I was working around there, but I've removed the extra fuss in https://github.com/jverzani/SpecialPolynomials.jl/pull/25